packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -48,6 +50,17 @@ export namespace WorkspaceTextEdit {
     }
 }
 
+export namespace CustomEditorWidget {
+    export function is(arg: Widget | undefined): arg is CustomEditorWidget {
+        return !!arg && 'modelRef' in arg;
+    }
+}
+
+export interface CustomEditorWidget extends Widget {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    readonly modelRef: Reference<any>;
```

Comment:
I'm not sure that `monaco-workspace.ts` is the best place to include this information (`CustomEditorWidget` namespace and interface)

Reply:
Moved it to **editor** (packages\editor\src\browser\editor.ts)